app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -9,31 +9,94 @@ module ClassMethods
     def validate_config_info(options)
       config_info = options[:config_info]
 
+      # ------------------------------------------------------------------------
+      # TransformationMapping validation
+      # ------------------------------------------------------------------------
       mapping = if config_info[:transformation_mapping_id]
                   TransformationMapping.find(config_info[:transformation_mapping_id])
                 else
                   config_info[:transformation_mapping]
                 end
-
+      # mapping.invalid? added as a part of validating migration plan
       raise _('Must provide an existing transformation mapping') if mapping.blank?
+      unless mapping.valid?
+        raise _('Transformation mapping was invalid.  ' + mapping.errors.messages)
+      end
```

Comment:
Typically something like this we would one-line, e.g. `raise _('Transformation mapping was invalid.  ' + mapping.errors.messages) unless mapping.valid?`

Also if possible `if mapping.invalid?` is preferred over `unless mapping.valid?`